sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -1735,12 +1829,28 @@ lsc_scatter(AccessorTy acc, __ESIMD_NS::simd<uint32_t, N> offsets,
   using _CstT = typename detail::lsc_bitcast_type<T>::type;
   __ESIMD_NS::simd<MsgT, N * NElts> Tmp = vals.template bit_cast_view<_CstT>();
   auto si = __ESIMD_NS::get_surface_index(acc);
+  auto loc_offsets = convert<uint32_t>(offsets);
```

Comment:
This convert seems redundant. offsets is vector of uint32_t here.
Please check other similar places too.